solr/solr-ref-guide/src/common-query-parameters.adoc

```diff
@@ -206,7 +206,7 @@ The default value of this parameter is blank, which causes no extra "explain inf
 
 == timeAllowed Parameter
 
-This parameter specifies the amount of time, in milliseconds, allowed for a search to complete. If this time expires before the search is complete, any partial results will be returned, but values such as `numFound`, <<faceting.adoc#faceting,facet>> counts, and result <<the-stats-component.adoc#the-stats-component,stats>> may not be accurate for the entire result set. In case of expiration, if `omitHeader` isn't set to `true` the response header contains a special flag called `partialResults`.
+This parameter specifies the amount of time, in milliseconds, allowed for a search to complete. If this time expires before the search is complete, any partial results will be returned, but values such as `numFound`, <<pagination-of-results.adoc#using-cursors,`nextCursorMark`>>, <<faceting.adoc#faceting,facet>> counts, and result <<the-stats-component.adoc#the-stats-component,stats>> may not be accurate for the entire result set. In case of expiration, if `omitHeader` isn't set to `true` the response header contains a special flag called `partialResults`.
```

Comment:
I don't think it's correct to say that nextCursorMark will be inaccurate. It will be accurate for the next result that should be returned.

This line also makes me think that maybe `timeAllowed+cursorMark+omitHeader` shouldn't be a valid combination. It's not a regression because `timeAllowed+cursorMark` is already disallowed. WDYT?

Reply:
I was just adding to the existing list there.  I'll pull it out into its own sentence.

In my opinion, `timeAllowed+cursorMark+omitHeader` should be allowed, and SOLR shouldn't be second-guessing the caller's request:

1. The combination `shards.tolerant+cursorMark` has the same issue, and `shards.tolerant+cursorMark+omitHeader` is allowed.
2. There may be a reasonable scenario that simply doesn't call for it, even if it is advisable to keep the header.  Let's say you are implementing a product catalog, and:
    - For whatever reason, you don't need the header, you rely on the http status code
    - You want to use cursorMarks to allow people to scroll through your fine list of shoes or whatever
    - You want to time-bound that resultset and protect your infrastructure from long-running queries
    - It just simply isn't mission critical for that every user to see every shoe.  If some rare expensive query or latency-producing event occurs, and truncates your shoe list, it is okay (you probably wouldn't bother retrying if you knew partialResults were true anyway).

My feeling is also a bit of a reaction to being annoyed about SolrCloud preventing us from creating collections without replicas, because we had legitimate reasons for standing up the replicas ourselves (thankfully, there is a workaround).


Then again, I say this as someone who doesn't have to answer Solr customer questions, so my words are just words.

Reply:
Sure, if it's clearly documented that certain combinations of parameters lead to unknowable results, then I think it's fine. Maybe visit the `omitHeader` docs and put a blatant warning there about use in conjunction with shards.tolerant, timeAllowed, or any other partial result operations. (WAND?)